lib/webauth.js

```diff
@@ -295,6 +295,37 @@ module.exports = function(authManager) {
       }
       res.status(200).send('');
     },
+
+    semiAuthenticatedMiddleware: Promise.coroutine(function*(req, res, next) {
+      try {
+        const isWebsocket = req.url.endsWith(".websocket");
+        const handler = getAuthHandler(req, authManager);
+        if (!handler) {
+          res.status(401).send("Authentication failed: auth type missing")
+          return;
+        }
+        const authPluginID = handler.pluginID;
+        let result;
+        const timeout = req.session.zlux ? req.session.zlux.expirationTime : 0;
+        if (authManager.sessionTimeoutMs !== TIMEOUT_VALUE_NO_EXPIRE
+            && (!timeout || timeout < Date.now())) {
+              req.session.zlux = undefined;
+              result = {authenticated: false, authorized: false};
+        } else {
+          const authPluginSession = getAuthPluginSession(req, authPluginID, {});
```

Comment:
No else here, this does authorize() and we don't care for semi-auth.
In the IF above, it should do a `res.status(401).json(result); return;`
Otherwise, `next(); return;` means success

Reply:
adding the res.status(401) breaks the login page since its querying it without authentication 

removing the else breaks logging in to a user with allwoed plugins, the allowed plugins doesn't show